src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
```

Comment:
Make sure that your exception messages are specific to the CountryDao. The message 'Cannot save country: %s to DB' is correct, but it's always a good practice to double-check that exception messages are meaningful and specific to the context they are used in.